### DISCUSSION TITLE
posts/develop-a-neovim-plugin-in-lua/

### DISCUSSION THREAD

**Discussion by giscus** [m4xshen/m4xshen.dev#7] [Announcements]
# posts/develop-a-neovim-plugin-in-lua/

Learn how to develop a Neovim plugin in Lua. Understand the structure of Neovim plugin, Lua module, and create a simple plugin.

https://m4xshen.dev/posts/develop-a-neovim-plugin-in-lua/

**Comment by pauladam94** [comment]
Very cool post. This really helps !

However I would prefer an extended example. I think one major thing that people want to do with a custom plugin is to change what is display in the current buffer.

Maybe an example on how to change "-" to small point  in a markdown file will be pertinent ?

**Reply by m4xshen** [reply]
Thank you! However, I would like to keep the example simple to follow. If you want to see an extended example, you can check out some simple plugins such as [mini.nvim](https://github.com/echasnovski/mini.nvim). For instance, [this one](https://github.com/echasnovski/mini.nvim/blob/main/lua/mini/cursorword.lua) spans within 300 lines, and there are also detailed comments explaining what it does.

**Reply by pauladam94** [reply]
Yes, I understand perfectly why you are keeping it minimal. Thanks for the link to mini.vim.

**Comment by kqvanity** [comment]
Nice article.
I'm wondering if there's a 'reactive' streamlined way to hot-reload changes, especially when working with the editor's various APIs.

**Reply by m4xshen** [reply]
Thanks! You can try the solution in [this reddit comment](https://www.reddit.com/r/neovim/comments/170jkzh/comment/k3kyf1k/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button).

**Comment by resoliwan** [comment]
Thank you, it has been very helpful to me.
